test/connectivity/modules/NetworkController/LinuxFirewallOfflineController.cs

```diff
@@ -0,0 +1,122 @@
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LinuxFirewallOfflineController>();
+        readonly string networkInterfaceName;
+        readonly string iotHubHostname;
+
+        public LinuxFirewallOfflineController(string networkInterfaceName, string iotHubHostname)
+        {
+            this.networkInterfaceName = networkInterfaceName;
+            this.iotHubHostname = iotHubHostname;
+        }
+
+        public string Description => "LinuxFirewallOffline";
+
+        public async Task<NetworkStatus> GetStatus(CancellationToken cs)
+        {
+            try
+            {
+                string output = await CommandExecutor.Execute(
+                    "tc",
+                    $"qdisc show dev {this.networkInterfaceName}",
+                    cs);
+
+                // parse output to see if online or offline
+                if (output.Contains("qdisc noqueue"))
+                {
+                    return NetworkStatus.Default;
+                }
+                else
+                {
+                    return NetworkStatus.Restricted;
+                }
+            }
+            catch (Exception e)
+            {
+                Log.LogError(e, "Failed to get network status");
+                return NetworkStatus.Unknown;
+            }
+        }
+
+        public Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
+        {
+            switch (status)
+            {
+                case NetworkStatus.Restricted:
+                    return this.AddDropRule(cs);
+                case NetworkStatus.Default:
+                    return this.RemoveDropRule(cs);
+                default:
+                    throw new NotSupportedException($"Set status '{status}' is not supported.");
+            }
+        }
+
+        async Task<bool> RemoveDropRule(CancellationToken cs)
+        {
+            try
+            {
+                // Delete the root rules cleans the chilldren rules
+                await CommandExecutor.Execute(
+                   "tc",
+                   $"qdisc delete dev {this.networkInterfaceName} root",
+                   cs);
+
+                return true;
+            }
+            catch (Exception e)
+            {
+                Log.LogError(e, "Failed to set accept rule");
+                return false;
+            }
+        }
+
+        async Task<bool> AddDropRule(CancellationToken cs)
+        {
+            try
+            {
+                IPAddress[] iothubAddresses = await Dns.GetHostAddressesAsync(this.iotHubHostname);
+                if (iothubAddresses.Length == 0)
+                {
+                    throw new CommandExecutionException("No IP found for iothub hostname");
+                }
+
+                foreach (var item in iothubAddresses)
+                {
+                    Log.LogInformation($"Found iotHub IP {item}");
+                }
+
+                await CommandExecutor.Execute(
```

Comment:
can you explain about below few await calls? what does these do?  I expect this method should be reused for satellite scenario, right?